### ISSUE TITLE
DB table creation error in 1.5.6 

### ISSUE THREAD

**Issue by scottcwilson** [zencart/zencart#1740]
Session table creation fails in MySQL 5.7.21.

```sql
CREATE TABLE sessions (
  sesskey varchar(191) default NULL,
  expiry int(11) unsigned NOT NULL default 0,
  value mediumblob NOT NULL,
  PRIMARY KEY  (sesskey)
) ENGINE=InnoDB
```
MySQL said: `#1171 - All parts of a PRIMARY KEY must be NOT NULL; if you need NULL in a key, use UNIQUE instead`

**Comment by scottcwilson** [comment]
I'm not sure how to fix this because I'm not sure of the original intent of the change in the sesskey field (from 255 to 191, from NOT NULL to NULL).  But we could certainly go to back to 

`sesskey varchar(255) NOT NULL default '',
`
The odd thing is that the upgrade script uses a different field definition: 

`  sesskey varchar(256) default NULL,
`
@drbyte let me know what you want here and I'll be glad to do the PR.

**Comment by scottcwilson** [comment]
What if we were to use the original 256 number for sessions and whos_online, and say 1.5.6 requires MySQL 5.7, with a note saying  drop this to 255 if you are using MySQL 5.6?

**Comment by drbyte** [comment]
Your error message is related to the NULL vs NOT NULL change.

The 256 vs 255 vs 191 is unrelated to the error message. But the ability to index properly for longer data is impacted if utf8mb4 is used, hence the move to 191.

As for the upgrade def I think I probably used 256 just for consistency across the various tables. I think it actually only supports 255 for varchar anyway. 

Evidently it still requires further changes.
And also extensive testing.

**Comment by drbyte** [comment]
Closed via #1763

**Comment by scottcwilson** [comment]
09/12/18 - tested in 1.5.6 both creation from scratch and upgrade; both worked perfectly!
